src/main/java/com/rabbitmq/http/client/domain/DetailsParameters.java

```diff
@@ -0,0 +1,108 @@
+ * own parameters applied on it, see {@link #DetailsParameters(QueryParameters)} and {@link
+ * #withQueryParameters()}. This way {@link DetailsParameters} can be "injected" in methods like
+ * {@link com.rabbitmq.http.client.Client#getQueues(DetailsParameters)}.
+ *
+ * @since 4.1.0
+ * @see com.rabbitmq.http.client.Client#getQueues(DetailsParameters)
+ * @see com.rabbitmq.http.client.Client#getQueues(QueryParameters)
+ * @see com.rabbitmq.http.client.Client#getQueue(String, String, DetailsParameters)
+ * @see com.rabbitmq.http.client.ReactorNettyClient#getQueues(String, DetailsParameters)
+ * @see com.rabbitmq.http.client.ReactorNettyClient#getQueue(String, String, DetailsParameters)
+ */
+public class DetailsParameters {
+
+  private final Map<String, String> parameters = new HashMap<>();
+  private final QueryParameters queryParameters;
+
+  public DetailsParameters() {
+    this(null);
+  }
+
+  public DetailsParameters(QueryParameters queryParameters) {
+    this.queryParameters = queryParameters;
+  }
+
+  private static void checkGreaterThanZero(int value, String field) {
+    if (value <= 0) {
+      throw new IllegalArgumentException(String.format("'%s' must be greater than 0", field));
+    }
+  }
+
+  private static void checkGreaterThanZero(Duration value, String field) {
+    if (value == null) {
+      throw new IllegalArgumentException(String.format("'%s' cannot be null", field));
+    }
+    if (value.toSeconds() <= 0) {
+      throw new IllegalArgumentException(String.format("'%s' must be greater than 0", field));
+    }
+  }
+
+  public DetailsParameters messageRates(int ageSeconds, int incrementSeconds) {
+    checkGreaterThanZero(ageSeconds, "age");
+    checkGreaterThanZero(incrementSeconds, "increment");
+    this.parameters.put("msg_rates_age", String.valueOf(ageSeconds));
+    this.parameters.put("msg_rates_incr", String.valueOf(incrementSeconds));
+    return this;
+  }
+
+  public DetailsParameters messageRates(Duration age, Duration increment) {
+    checkGreaterThanZero(age, "age");
+    checkGreaterThanZero(increment, "increment");
+    return this.messageRates((int) age.toSeconds(), (int) increment.toSeconds());
+  }
+
+  public DetailsParameters lengths(int ageSeconds, int incrementSeconds) {
+    checkGreaterThanZero(ageSeconds, "age");
+    checkGreaterThanZero(incrementSeconds, "increment");
+    this.parameters.put("lengths_age", String.valueOf(ageSeconds));
+    this.parameters.put("lengths_incr", String.valueOf(incrementSeconds));
+    return this;
+  }
+
+  public DetailsParameters lengths(Duration age, Duration increment) {
+    checkGreaterThanZero(age, "age");
+    checkGreaterThanZero(increment, "increment");
+    return this.lengths((int) age.toSeconds(), (int) increment.toSeconds());
+  }
+
+  public Map<String, String> parameters() {
+    return new HashMap<>(parameters);
+  }
+
+  public QueryParameters withQueryParameters() {
```

Comment:
Rather than remembering to call this method to fill in the queryparameters, do you think what I suggest below could be an alternative?

## Do not take QueryParameter in the constructor instead via a separate method
```
DetailParameters() {
   
}
public QueryParameters queryParameters() {
  if (queryParameters == null) {
     queryParameters = new QueryParameters(this.parameters);
  }
  return queryParameters.
}

```
Only when we need to specify queryParameters we ask for it (`queryParameters()`) and an instance is created -under the covers- if this is the first time it is called. Any property set on the queryParameter is actually set in the parameters of the DetailParameters' map as we can see in the next section.
```
parameters = new DetailParameters()
parameters.queryParameters()
    .name("....")
    .pagination()..
    ...
```

## Have a package level protected constructor for QueryParameters
The actual parameters map is provided by the DetailParameters. They all live on the same package and this constructor can be protected with package level.
```
package-level QueryParameters(Map<String,String> parameters) {
   this.parameters = parameters;
}
```